src/signMessage.c

```diff
@@ -167,22 +165,83 @@ void handle_sign_message_parse_message(volatile unsigned int *tx) {
     }
 }
 
+static bool check_swap_validity(const SummaryItemKind_t kinds[MAX_TRANSACTION_SUMMARY_ITEMS],
```

Comment:
You don't think this function should be defined in `handle_swap_sign_transaction` ? @fbeutin-ledger

Reply:
I didn't want to move the handling of the UI formatted fields to `handle_swap_sign_transaction` that's why I've put it here and the `check_swap_amount` / `check_swap_recipient` on swap code side
However I agree with your comment that more stuff should go to `handle_swap_sign_transaction`, I've transfered code in the `check_xyz` let me know if it's sufficient

Reply:
It looks better to me. I still think it would be better to keep all things swap related in one place but I understand your point regarding not moving the field formatting...

I'm resolving the conversation.